bitvm/src/hash/sha256_test_utils.rs

```diff
@@ -0,0 +1,67 @@
+use crate::hash::sha256::reference_sha256;
+use serde::Deserialize;
+use sha2::Digest;
+use std::error::Error;
+use std::fs::File;
+use std::io::BufReader;
+
+#[derive(Deserialize, Debug)]
+pub struct Message {
+    pub data: String,
+    pub count: u64,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct TestVector {
+    pub message: Message,
+    #[serde(rename = "SHA-256")]
+    pub sha256: String,
+}
+
+/// Read test vectors from JSON file
+pub fn read_sha256_test_vectors(
+    input_limit: usize,
+) -> Result<Vec<(String, String)>, Box<dyn Error>> {
+    let path = "src/hash/sha256_official_test_vectors.json";
+    let file = File::open(path)?;
+    let reader = BufReader::new(file);
+    let (_ignored, vectors): (serde::de::IgnoredAny, Vec<TestVector>) =
+        serde_json::from_reader(reader)?;
+
+    let mut prepared_vectors = vec![];
+    for vector in vectors.iter() {
+        let input = vector
+            .message
+            .data
+            .repeat(vector.message.count as usize)
+            .as_bytes()
+            .to_vec();
+        if input.len() > input_limit {
+            // The implementation only supports inputs up to 128 bytes
```

Comment:
This limit is wrong, no? It depends on the implementation, hence it being a function parameter.